spec/support/govuk_i18n_cov.rb

```diff
@@ -0,0 +1,81 @@
+# frozen_string_literal: true
+
+class GOVUKI18nCov
+  module I18n
+    module Registry
+      def lookup(locale, key, scope = [], options = {})
+        GOVUKI18nCov.log(locale, key, scope)
+        super
+      end
+    end
+  end
+
+  class << self
+    def start
+      return if @started
+
+      @started = true
+      @used_keys = []
+      ::I18n::Backend::Simple.send :include, I18n::Registry
+    end
+
+    def log(locale, key, scope)
+      prefix = [locale] + Array(scope || [])
+      @used_keys << "#{prefix.compact.join('.')}.#{key}"
+    end
+
+    def report
+      return unless @started
+
+      used = normalise_keys(@used_keys).uniq
+      all = get_all_i18n_keys
+      unused = all.reject { |as| used.any? { |us| as.start_with?(us) } }
+
+      if unused.any?
+        write_report(unused)
+        print_message(all, unused)
+      end
+    end
+
+  private
+
+    def get_all_i18n_keys
+      ymls = Dir.glob("config/locales/**/*.yml")
+      locales = ymls.map { |yml| YAML.load_file(yml) }.reduce({}, :deep_merge)
+
+      locales.flat_map { |key, value| key_chains_for(key, value) }
+        .map { |key_chain| key_chain.join(".") }
+    end
+
+    def key_chains_for(key, value)
+      return [[key]] unless value.is_a? Hash
+
+      value.flat_map { |sub_key, sub_value| key_chains_for(sub_key, sub_value) }
+        .map { |sub_key| [key] + sub_key }
+    end
+
+    def normalise_keys(keys)
+      keys.uniq.map(&:to_s).map { |s| s.sub(/^\./, "") }
+    end
+
+    def write_report(unused)
+      FileUtils.mkdir_p("coverage")
+
+      File.open("coverage/i18n", "w") do |f|
```

Comment:
I have a recollection that you said this file was in the same format as simplecov, hopefully I'm not making that up. I notice this is a list but simplecov drops out a HTML file.